File: backend-go/internal/dto/price.go
package dto

type PriceRequest struct {
	AssetCategory string  `json:"assetCategory" binding:"required"`
	AssetName     string  `json:"assetName" binding:"required"`
	Specification string  `json:"specification"`
	Unit          string  `json:"unit"`
	BasePrice     float64 `json:"basePrice" binding:"required"`
	EffectiveDate string  `json:"effectiveDate"`
	ExpiryDate    string  `json:"expiryDate"`
	Remark        string  `json:"remark"`
}

type PriceResponse struct {
	ID            int64   `json:"id"`
	AssetCategory string  `json:"assetCategory"`
	AssetName     string  `json:"assetName"`
	Specification string  `json:"specification"`
	Unit          string  `json:"unit"`
	BasePrice     float64 `json:"basePrice"`
	EffectiveDate *string `json:"effectiveDate"`
	ExpiryDate    *string `json:"expiryDate"`
	Remark        string  `json:"remark"`
}

File: frontend/src/api/price.ts
import request from '../utils/request'

export interface PriceRecord {
  id: number
  assetCategory: string
  assetName: string
  specification?: string | null
  unit?: string | null
  basePrice: number
  effectiveDate?: string | null
  expiryDate?: string | null
  remark?: string | null
}

export interface PricePayload {
  assetCategory: string
  assetName: string
  specification?: string
  unit?: string
  basePrice: number
  effectiveDate?: string
  expiryDate?: string
  remark?: string
}

export function getPrices() {
  return request.get('/prices')
}

export function createPrice(payload: PricePayload) {
  return request.post('/prices', payload)
}

export function updatePrice(id: number, payload: PricePayload) {
  return request.put(`/prices/${id}`, payload)
}

export function deletePrice(id: number) {
  return request.delete(`/prices/${id}`)
}

export function lookupPrices(category: string, name?: string) {
  return request.get('/prices/lookup', { params: { category, name: name || undefined } })
}
